bpf/kprobe_pwru.c

```diff
@@ -393,6 +393,17 @@ PWRU_ADD_KPROBE(3)
 PWRU_ADD_KPROBE(4)
 PWRU_ADD_KPROBE(5)
 
+SEC("kprobe/skb_by_stackid")
+int kprobe_skb_by_stackid(struct pt_regs *ctx) {
+	u64 stackid = get_stackid(ctx);
+
+	struct sk_buff **skb = bpf_map_lookup_elem(&stackid_skb, &stackid);
+	if (skb && *skb)
+		return kprobe_skb(*skb, ctx, PWRU_HAS_GET_FUNC_IP);
```

Comment:
I should pass the `stackid` as an arguement of `kprobe_skb` so we don't have to get_stackid() for twice.